test/unit/csv_file_from_public_host_test.rb

```diff
@@ -0,0 +1,32 @@
+require "test_helper"
+
+class CsvFileFromPublicHostTest < ActiveSupport::TestCase
+  def stub_csv_request(status: 206, body: '')
+    stub_request(:get, "#{Whitehall.public_root}/some-path")
+      .with(headers: { 'Range' => 'bytes=0-30000' })
+      .to_return(status: status, body: body)
+  end
+
+  test '#new yields a temporary file' do
+    stub_csv_request
+
+    CsvFileFromPublicHost.new('some-path') do |file|
+      assert File.exist?(file.path)
+    end
+  end
+
+  test '#new yields a temporary file that contains the contents of the request body' do
+    stub_csv_request(body: 'csv,file')
+
+    CsvFileFromPublicHost.new('some-path') do |file|
+      assert_equal 'csv,file', file.read
+    end
+  end
+
+  test '#new raises an exception if the request status is anything other than 206' do
```

Comment:
Out of interest, do you still get a 206 if the file being requested is smaller than 30,000 bytes?

Reply:
Yes, it appears so. For example 

```
curl https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/675936/Q4_2017_csv.csv -H "Range: bytes=0-1000000" -v > /dev/null 
```

This file is 51798 bytes, but requesting the first 1000000 still returns a 206.